feat: service logs full download

Dartoxian opened the pull request:
## Description:
This PR implements the full streamed download of service logs. This is based on some work already done by @tedim52  in tedi/copydownload.

Changes:
* The copy logs button is no longer available on the Service Logs page
* The download logs button sends the log stream to a stream writer to download them as a file.

Concerns:
* For services that log excessively, like the log spammer, it seems downloading the logs is incredibly slow - <1kb/s. I'm not sure what could improve this, but if the backend streamed more than one record per message (as seems to be possible by the api spec) that seems like it would help (and would help improve performance when rendering service logs in the ui).

### Demo

https://github.com/kurtosis-tech/kurtosis/assets/4419574/61716034-fb6e-47c1-b776-e8c262b40799

## Is this change user facing?
YES

Dartoxian committed:
Implement download full logs for service logs

Dartoxian committed:
Wrap the download in try/catch in case it fails

adschwartz commented on enclave-manager/web/src/components/DownloadButton.tsx:
```diff
@@ -1,10 +1,11 @@
 import { Button, ButtonProps, IconButton, IconButtonProps } from "@chakra-ui/react";
 import { FiDownload } from "react-icons/fi";
-import { isDefined } from "../utils";
+import streamsaver from "streamsaver";
```
@Dartoxian do you need to add this guy to `package.json`?

Dartoxian replied:
no it's already there.

adschwartz commented on enclave-manager/web/src/components/DownloadButton.tsx:
```diff
@@ -17,9 +18,20 @@ export const DownloadButton = <IsIconButton extends boolean>({
       const v = typeof valueToDownload === "string" ? valueToDownload : valueToDownload();
+
+      if (isAsyncIterable(v)) {
+        const writableStream = streamsaver.createWriteStream(fileName);
```
@Dartoxian I think there's a good chance this may work fine, but I wonder if you've tried downloading from your cloud instance? It's possible this might not work going over the proxies (without some tweaking) but it's possible it will.

Dartoxian replied:
Yep I've verified that the `streamsaver` library works in the cloud as it is also used to download artifact tgz files in the cloud.

Dartoxian committed:
Merge branch 'main' into bgazzard/service-logs-full-download

Dartoxian committed:
fix: emui small design tweaks based on team feedback (#1897)

## Description:
This PR includes a number of the easier emui tweaks from the december
burndown:
* Don't show copy button next to the package source

![image](https://github.com/kurtosis-tech/kurtosis/assets/4419574/9b598336-b4f4-4793-96c5-5a1b99a3ba48)
* Make the delete button have a hover

![image](https://github.com/kurtosis-tech/kurtosis/assets/4419574/8f391477-c07f-4271-a9b9-cec848cfbde3)
* ellipsis long enclave names on the enclaves list

![image](https://github.com/kurtosis-tech/kurtosis/assets/4419574/2429b5de-63e8-492a-84fc-e67c9da7072e)
* Don't include entrypoint and returns on the catalog package page
* Dynamic package card layout

![image](https://github.com/kurtosis-tech/kurtosis/assets/4419574/e3f902fc-7d03-4032-bc24-83517f8f8cf2)
* support navigating logs with enter and shift+enter when searching
* fix an issue with log search that was causing excessive renders, which
I think was causing the `ok.` bug mentioned on notion.

## Is this change user facing?
YES

## References (if applicable):
- December burndown notion

Dartoxian committed:
Build

adschwartz committed:
Merge branch 'main' into bgazzard/service-logs-full-download
